core/src/apps/wallet/sign_tx/addresses.py

```diff
@@ -214,21 +214,29 @@ def validate_full_path(
     ):
         return False
 
-    if path[1] != coin.slip44 | HARDENED:
+    if path[1] > 20 and path[1] != coin.slip44 | HARDENED:
         return False
-    if path[2] < HARDENED or path[2] > 20 | HARDENED:
+    if (path[2] > 20 and path[2] < HARDENED) or path[2] > 20 | HARDENED:
+        return False
+    if path[3] not in (0, 1, 0 | HARDENED, 1 | HARDENED, 2 | HARDENED):
```

Comment:
Is it correct that we restrict path[3] (address index for BIP45) only to 0' - 2'? The spec says `Addresses are numbered from index 0 in sequentially increasing manner. This number is used as child index in BIP32 derivation.`

Reply:
Correct. This range is again for Electrum which uses BIP48 in a weird manner.